src/encode/pattern/mod.rs

```diff
@@ -135,17 +135,131 @@ use crate::config::{Deserialize, Deserializers};
 
 mod parser;
 
+const DEFAULT_PATTERN_ENCODER: &str = "{d} {l} {t} - {m}{n}";
+
+#[allow(dead_code)]
+fn default_pattern() -> Option<String> {
+    Some(DEFAULT_PATTERN_ENCODER.to_owned())
+}
+
+/// A simple color map struct
+///
+/// You can use this struct to define a custom color map.
+/// This can be done using a serializer (i.e. config file) or pragmatically.
+///
+/// It is part of the encoder
+///
+/// ```not_rust
+///   pattern: "%m"
+///   color_map:
+///     INFO: Blue
+///     TRACE: Black
+/// ```
+///
+/// Or progmatically in conjuction with PatternEncoder
+///! ```no_run
+///! # #[cfg(all(feature = "console_appender",
+///! #           feature = "file_appender",
+///! #           feature = "pattern_encoder"))]
+///! # fn f() {
+///! let mut log_color_map = ColorMap::default()
+///! log_color_map.info_color = Some(log4rs::encode::Color::Cyan);
```

Comment:
Could not get that to work: `info_color` does not exists, and members seems private anyway here